daphne/src/roles.rs

```diff
@@ -529,43 +577,69 @@ where
         // Prepare the Leader's aggregate share.
         let leader_enc_agg_share = task_config.vdaf.produce_leader_encrypted_agg_share(
             &task_config.collector_hpke_config,
-            &collect_req.task_id,
+            task_id,
             &batch_selector,
             &leader_agg_share,
             task_config.version,
         )?;
 
         // Prepare AggregateShareReq.
         let agg_share_req = AggregateShareReq {
-            task_id: collect_req.task_id.clone(),
+            draft02_task_id: task_id.for_request_payload(&task_config.version),
             batch_sel: batch_selector.clone(),
             agg_param: collect_req.agg_param.clone(),
             report_count: leader_agg_share.report_count,
             checksum: leader_agg_share.checksum,
         };
 
+        let url = if task_config.version == DapVersion::Draft02 {
+            "aggregate_share".to_string()
+        } else {
+            format!("tasks/{}/aggregate_shares", task_id.to_base64url())
+        };
+
         // Send AggregateShareReq and receive AggregateShareResp.
         let resp = leader_post!(
             self,
-            &collect_req.task_id,
+            task_id,
             task_config,
-            "aggregate_share",
+            &url,
             MEDIA_TYPE_AGG_SHARE_REQ,
-            agg_share_req.get_encoded_with_param(&task_config.version)
+            DapResource::Undefined,
+            agg_share_req.get_encoded_with_param(&task_config.version),
+            false
         );
-        let agg_share_resp = AggregateShareResp::get_decoded(&resp.payload)?;
+        let agg_share_resp = AggregateShare::get_decoded(&resp.payload)?;
+        let interval: Option<Interval> = match task_config.version {
```

Comment:
I think it's worth commenting on what this code accomplishes.
```suggestion

        /// The Collection message includes the smallest quantized time interval containing all reports in the batch.
        let interval: Option<Interval> = match task_config.version {
```